knox/views.py

```diff
@@ -38,20 +41,29 @@ def post(self, request, format=None):
                     {"error": "Maximum amount of tokens allowed per user exceeded."},
                     status=status.HTTP_403_FORBIDDEN
                 )
+        show_token_ttl = self.get_ttl_visibility()
         token_ttl = self.get_token_ttl()
         token = AuthToken.objects.create(request.user, token_ttl)
         user_logged_in.send(sender=request.user.__class__,
                             request=request, user=request.user)
         UserSerializer = self.get_user_serializer_class()
+
         if UserSerializer is None:
-            return Response({
+            data = {
+                'token': token
+            }
+        else:
+            context = self.get_context()
+            data = {
+                'user': UserSerializer(
+                    request.user,
+                    context=context
+                ).data,
                 'token': token
-            })
-        context = self.get_context()
-        return Response({
-            'user': UserSerializer(request.user, context=context).data,
-            'token': token,
-        })
+            }
+        if show_token_ttl:
+            data['token_expiry'] = token_ttl
```

Comment:
oh nooo, you are right I completely misread that part. I will scamble a fix.

Reply:
@belugame sounds good, I am in favor of re-writing those parts to comply with "what is expected" since it is afterall a django project. @johnraz would you agree or do you want another direction?

Reply:
> What is the main reasoning to we not simply do the what django does and return the model instance? It gives alot more flexibility, no need to even return a tuple of the "instance + a duplicate part of the instance"?

The token, the string, is not stored as is in the `AuthToken`  but it's hashed with a salt.
That's why we need the tuple. 

> Adding new methods to circumvent old oddities is counterproductive I feel.
I do agree, so I would say that we should implement the "tuple" like behavior in the existing create method. Using create without returning the token is non-sense anyway

Reply:
@johnraz Yes, I just realized that, sorry for the noise. I could just overwrite the `instance.token_key` with the raw `token` before returning it thus getting best of both worlds (simplicity and the functionality we need). Ill type something up, ill have it ready in a bit.

I agree with you and @belugame I cannot find a good reason to keep the setting for this, and just always include it unless the expiration is set to `None` then obviously we dont return it.

Reply:
I would always return the `token_ttl`, if it's `None` it will be `null` in the payload and that's fine. Otherwise frontend libraries would need to check for existence etc.

I wouldn't erase the value of `instance.token_key` because it will make things confusing and implicit, to quote the zen of Python "Explicit is better than implicit" and I think the most explicit solution is to return a tuple. It changes the default return value of a create method which is not ideal but at the same time it is not hidden and one would easily spot that the method returned a tuple by reading our code:

Override the `token_key`
```
instance = AuthToken.objects.create() <-- no sign of an overridden token_key here ; implicit behavior :/
```

Tuple
```
instance, token = AuthToken.objects.create() <-- explicitly assign the token which is more readable.
```

Reply:
@johnraz you are absolutly right, I am moving to quickly. I ended up implementing what you said. Tossed the setting and made the create method return the instance and token :smiley_cat: